posthog/queries/query_date_range.py

```diff
@@ -221,4 +247,6 @@ def should_round(self):
         return round_interval
 
     def is_hourly(self, target):
-        return self._filter.interval == "hour" or (target and isinstance(target, str) and "h" in target)
+        return getattr(self._filter, "interval", None) == "hour" or (
```

Comment:
Needing to use getattr with filters makes the code much harder to work with. Can we avoid this somehow?

Reply:
This is just duck typing – it was only by happenstance that previously all filter types included `IntervalMixin`, but that isn't fundamentally required for `QueryDateRange`, and indeed `PropertiesTimelineFilter` has no `interval`. Alternatively this could be an `if not isinstance(self._filter, IntervalMixin): return False` guard. What do you think?